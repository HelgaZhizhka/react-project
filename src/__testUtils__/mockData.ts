import { Photo, SearchResponse } from '@/types/interfaces';

export const PhotoItem: Photo = {
  id: 247600,
  width: 4288,
  height: 2704,
  url: 'https://www.pexels.com/photo/scenic-view-of-lake-in-forest-247600/',
  photographer: 'Pixabay',
  photographer_url: 'https://www.pexels.com/@pixabay',
  photographer_id: 2659,
  avg_color: '#426E6B',
  src: {
    original:
      'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200',
    large2x:
      'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=650&w=940',
    large:
      'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940',
    medium:
      'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=350',
    small: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg',
    portrait:
      'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800',
    landscape:
      'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=130',
    tiny: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
  },
  alt: 'Scenic View of Lake in Forest',
};
export const PhotoData = [
  {
    id: 247601,
    width: 4288,
    height: 2704,
    url: 'https://www.pexels.com/photo/scenic-view-of-lake-in-forest-247600/',
    photographer: 'Pixabay',
    photographer_url: 'https://www.pexels.com/@pixabay',
    photographer_id: 2659,
    avg_color: '#426E6B',
    src: {
      original:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200',
      large2x:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=650&w=940',
      large:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940',
      medium:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=350',
      small: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg',
      portrait:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800',
      landscape:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=130',
      tiny: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
    },
    alt: 'Scenic View of Lake in Forest',
  },
  {
    id: 247602,
    width: 4288,
    height: 2704,
    url: 'https://www.pexels.com/photo/scenic-view-of-lake-in-forest-247600/',
    photographer: 'Pixabay',
    photographer_url: 'https://www.pexels.com/@pixabay',
    photographer_id: 2659,
    avg_color: '#426E6B',
    src: {
      original:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200',
      large2x:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=650&w=940',
      large:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940',
      medium:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=350',
      small: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg',
      portrait:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800',
      landscape:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=130',
      tiny: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
    },
    alt: 'Scenic View of Lake in Forest',
  },
  {
    id: 247603,
    width: 4288,
    height: 2704,
    url: 'https://www.pexels.com/photo/scenic-view-of-lake-in-forest-247600/',
    photographer: 'Pixabay',
    photographer_url: 'https://www.pexels.com/@pixabay',
    photographer_id: 2659,
    avg_color: '#426E6B',
    src: {
      original:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200',
      large2x:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=650&w=940',
      large:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940',
      medium:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=350',
      small: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg',
      portrait:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800',
      landscape:
        'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&h=130',
      tiny: 'https://images.pexels.com/photos/247600/pexels-photo-247600.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
    },
    alt: 'Scenic View of Lake in Forest',
  },
];

export const ResponseData: SearchResponse = {
  page: 1,
  per_page: 10,
  photos: PhotoData,
  total_results: 124,
  prev_page: '',
  next_page: '',
};
