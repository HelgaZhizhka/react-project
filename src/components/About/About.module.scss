.root {
  position: fixed;
  top: 0;
  right: 0;
  height: 100vh;
  width: 300px;
  overflow-y: auto;
  z-index: 100;
}

.container {
  min-height: 100%;
  padding: 50px 20px 20px;
  background-color: var(--component-bg);
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
}

.close {
  position: absolute;
  top: 5px;
  right: 20px;
  z-index: 10;
}

.overlay {
  position: fixed;
  top: 0;
  right: 0;
  height: 100vh;
  width: 100vw;
  background-color: rgba(0, 0, 0, 0.1);
  z-index: 99;
}