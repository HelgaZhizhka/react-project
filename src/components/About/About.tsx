import { useRouter } from 'next/router';

import { currentPage, defaultPerPage } from '@/utils/constants';
import { Routes } from '@/types/enums';
import { Button } from '@/components/Button';
import { Photo } from '@/types/interfaces';
import { Card } from '@/components/Card';
import styles from './About.module.scss';

type Props = {
  photoData: Photo;
};

const About: React.FC<Props> = ({ photoData }) => {
  const router = useRouter();
  const { query = '', page = currentPage, per_page = defaultPerPage } = router.query;

  const handleClose = () => {
    if (query) {
      router.push(`${Routes.HOME}?query=${query}&page=${page}&per_page=${per_page}`);
    } else {
      router.push(`${Routes.HOME}?page=${page}&per_page=${per_page}`);
    }
  };

  return (
    <aside className={styles.root}>
      <div className={styles.overlay} onClick={handleClose}></div>
      <div className={styles.container}>
        <Button className={styles.close} onClick={handleClose}>
          Close
        </Button>
        <Card {...photoData} isDetailed={true} />
      </div>
    </aside>
  );
};

export default About;
