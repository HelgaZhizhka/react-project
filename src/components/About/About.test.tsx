import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime';

import { PhotoItem } from '@/__testUtils__/mockData';
import createMockRouter from '@/__testUtils__/createMockRouter';
import About from './About';

describe('Details Component', () => {
  const mockRouter = createMockRouter({ query: { id: '247600' } });

  it('displays detailed card data correctly', async () => {
    render(
      <RouterContext.Provider value={mockRouter}>
        <About photoData={PhotoItem} />
      </RouterContext.Provider>
    );
    const closeButton = screen.getByText(/Close/i);

    await waitFor(() => {
      expect(closeButton).toBeInTheDocument();
    });
  });

  it('redirects to the home page when the close button is clicked', async () => {
    render(
      <RouterContext.Provider value={mockRouter}>
        <About photoData={PhotoItem} />
      </RouterContext.Provider>
    );

    const closeButton = screen.getByText(/Close/i);
    fireEvent.click(closeButton);

    await waitFor(() => {
      expect(mockRouter.push).toHaveBeenCalled();
    });
  });
});
