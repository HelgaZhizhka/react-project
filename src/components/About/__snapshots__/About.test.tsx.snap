// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component displays detailed card data correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="root"
      >
        <div
          class="overlay"
        />
        <div
          class="container"
        >
          <button
            class="root close"
            type="button"
          >
            Close
          </button>
          <div
            class="root detailed"
            role="listitem"
          >
            <div
              class="wrapper"
              style="background-color: rgb(66, 110, 107);"
            >
              <img
                alt="Scenic View of Lake in Forest"
                class="image"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="2704"
                src="/_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F247600%2Fpexels-photo-247600.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26dpr%3D2%26h%3D650%26w%3D940&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F247600%2Fpexels-photo-247600.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26dpr%3D2%26h%3D650%26w%3D940&w=3840&q=75 1x"
                style="color: transparent;"
                width="4288"
              />
            </div>
            <div
              class="content"
            >
              <p
                class="description"
              >
                Scenic View of Lake in Forest
              </p>
              <a
                href="https://www.pexels.com/@pixabay"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h2
                  class="title"
                >
                  Pixabay
                </h2>
              </a>
            </div>
          </div>
        </div>
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside
      class="root"
    >
      <div
        class="overlay"
      />
      <div
        class="container"
      >
        <button
          class="root close"
          type="button"
        >
          Close
        </button>
        <div
          class="root detailed"
          role="listitem"
        >
          <div
            class="wrapper"
            style="background-color: rgb(66, 110, 107);"
          >
            <img
              alt="Scenic View of Lake in Forest"
              class="image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="2704"
              src="/_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F247600%2Fpexels-photo-247600.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26dpr%3D2%26h%3D650%26w%3D940&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.pexels.com%2Fphotos%2F247600%2Fpexels-photo-247600.jpeg%3Fauto%3Dcompress%26cs%3Dtinysrgb%26dpr%3D2%26h%3D650%26w%3D940&w=3840&q=75 1x"
              style="color: transparent;"
              width="4288"
            />
          </div>
          <div
            class="content"
          >
            <p
              class="description"
            >
              Scenic View of Lake in Forest
            </p>
            <a
              href="https://www.pexels.com/@pixabay"
              rel="noopener noreferrer"
              target="_blank"
            >
              <h2
                class="title"
              >
                Pixabay
              </h2>
            </a>
          </div>
        </div>
      </div>
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
